version: '3.8'
services:
  mysql-db:
    image: mysql:8.0.29
    container_name: mysql-db
    ports:
      - "3308:3306"
    environment:
      - MYSQL_DATABASE=storefront
      - MYSQL_ALLOW_EMPTY_PASSWORD=yes
      
    command: --init-file /data/application/init.sql
    volumes:
      - ./init.sql:/data/application/init.sql
      - ./database:/var/lib/mysql
    healthcheck:
        test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
        timeout: 3s
        retries: 5
    networks: 
      - mysql-connection
  db-admin:    
    image: phpmyadmin:latest
    container_name: mysql-db-admin
    restart: always
    environment:
      PMA_HOST: mysql-db
      PMA_USER: root
    ports:      
      - "8080:80"
    depends_on:
      mysql-db:
        condition: service_healthy
    networks: 
      - mysql-connection
  storefront:
    build: 
      context: .
    container_name: storefront
    volumes:
      - .:/app/
    ports:
      - 5000:5000
    command: python manage.py runserver 0.0.0.0:5000
    depends_on:
      mysql-db:
        condition: service_healthy
    networks: 
      - mysql-connection
networks:
  mysql-connection: